### ISSUE TITLE
redirect failing in UserController::pageInfo when user not found

### ISSUE THREAD

**Issue by zbigcheese** [userfrosting/UserFrosting#888]
Recently it seems, named paths have stopped working.

Example route:

```
$app->group('/users', function () {
    $this->get('', 'UserFrosting\Sprinkle\Admin\Controller\UserController:pageList')
        ->setName('uri_users');
```

Example code from a controller:

```
$user = $this->getUserFromParams($args);
        // If the user no longer exists, forward to main user listing page
        if (!$user) {
            $usersPage = $this->ci->router->pathFor('uri_users');
            return $response->withRedirect($usersPage, 404);
        }
```

hitting the above point in a controller just produces a regular browser 404 page and does not redirect at all

**Comment by alexweissman** [comment]
It would seem that our use of a 404 status code in a redirect is not compliant with (HTTP?) specs. See https://github.com/slimphp/Slim/issues/2355

I'm not sure why we did this originally, and I can't imagine why 404 would ever have been an acceptable status code for redirects.  At any rate, we should probably just change the code to 302.

**Comment by lcharette** [comment]
Fixed in e7ac1d0 for 4.2

Tested and indeed 404 and 400 status code failed to redirect. Slim default to 302, so default it is now.
